hoptimator-operator/src/main/java/com/linkedin/hoptimator/operator/ControllerService.java

```diff
@@ -0,0 +1,27 @@
+package com.linkedin.hoptimator.operator;
+
+import org.apache.calcite.plan.RelOptRule;
+
+import io.kubernetes.client.extended.controller.Controller;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.ServiceLoader;
+import java.util.stream.Collectors;
+
+/** Loads controller plugins via the ControllerProvider Service Provider Interface. */
+public final class ControllerService {
+
+  public static Collection<ControllerProvider> providers() {
+    ServiceLoader<ControllerProvider> loader = ServiceLoader.load(ControllerProvider.class);
+    List<ControllerProvider> providers = new ArrayList<>();
+    loader.iterator().forEachRemaining(x -> providers.add(x));
+    return providers;
```

Comment:
nit: 
```java
// loader variable isn't required
// change providers to lambda
List<ControllerProvider> providers = new ArrayList<>(); 
ServiceLoader.load(ControllerProvider.class).iterator().forEachRemaining(providers::add);
return providers;
```

FI alternative (requires JavaCompile > 8).
```java
return ServiceLoader.load(ControllerProvider.class)
    .stream()
    .map(ServiceLoader.Provider::get)
    .collect(Collectors.toList());
```

Reply:
Looks like this depends on java 9, which is probably okay, but let's save that for a future PR. We can clean up a lot of small stuff like this if we skip to 11.